OrbitCore/LinuxTracingSession.cpp

```diff
@@ -26,12 +26,11 @@ void LinuxTracingSession::RecordHashedCallstack(
 bool LinuxTracingSession::ReadAllContextSwitches(
     std::vector<ContextSwitch>* buffer) {
   absl::MutexLock lock(&context_switch_buffer_mutex_);
-  if (context_swith_buffer_.empty()) {
+  if (context_switch_buffer_.empty()) {
     return false;
   }
 
-  std::swap(context_swith_buffer_, *buffer);
-  context_swith_buffer_.clear();
+  *buffer = std::move(context_switch_buffer_);
```

Comment:
You should leave  context_swith_buffer_.clear();  as if I'm not mistaken moving from leaves the object in a valid but unspecified state (which in this case is effectively empty but still). Same for the other similar methods.